core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -130,14 +131,16 @@ public int getAdvertisedPort() {
      * Gets the host name that we advertise protocol clients to connect to.
      * @since 2.198
      */
-    public String getAdvertisedHost() {
+    public String getAdvertisedHost() throws URISyntaxException {
```

Comment:
this is a breaking change.

Reply:
Removed  `throws URISyntaxException` and modified the block of code as below
```
public String getAdvertisedHost() {
        if (CLI_HOST_NAME != null) {
          return CLI_HOST_NAME;
        }
        try {
            return new URI(Jenkins.get().getRootUrl()).toURL().getHost();
        } catch (MalformedURLException | URISyntaxException e) {
            throw new IllegalStateException("Could not get TcpSlaveAgentListener host name", e);
        }
    }
```